docs/usage/dataclasses.md

```diff
@@ -57,6 +57,14 @@ them with `pydantic.dataclasses.dataclass`.
 ```
 _(This script is complete, it should run "as is")_
 
+Stdlib dataclasses (nested or not) can also be inherited and _pydantic_ will automatically validate
```

Comment:
we should also add a comment about the `arbitrary_types_allowed` logic change since that now behaves different from nested pydantic dataclasses or next pydantic models.